torch/autograd/graph.py

```diff
@@ -391,8 +392,9 @@ def inner_hook(grad: torch.Tensor):
     class Handle(RemovableHandle):
         handles: Tuple[RemovableHandle, ...]
 
-        def __init__(self, handles: Tuple[RemovableHandle, ...]):
+        def __init__(self, handles: Tuple[RemovableHandle, ...], grad_fns: Tuple[Node, ...]):
             self.handles = handles
+            self.grad_fns: Optional[Tuple] = grad_fns
```

Comment:
Regarding `Optional[Tuple]` typing, when can `grad_fns` be `None`?

Reply:
Whoops I meant to set it to None in `remove`, good catch